src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -75,8 +90,34 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
         public async Task<IPackageSearchMetadata> GetLatestPackageMetadataAsync(PackageIdentity identity,
             bool includePrerelease, CancellationToken cancellationToken)
         {
+            VersionRange allowedVersions = null;
+
+            // at this time, allowed version range should be applied at project level. 
+            // For solution level, we will apply allowed version range for all the selected projects in detail control model
+            if (!_isSolution)
+            {
+                if (_packageReferences == null)
+                {
+                    // get all package references for all the projects and cache locally
+                    var packageReferencesTasks = _projects.Select(project => project.GetInstalledPackagesAsync(cancellationToken));
+                    _packageReferences = (await Task.WhenAll(packageReferencesTasks)).SelectMany(p => p).Where(p => p != null);
+                }
+
+                // filter package references for current package identity
+                var matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.PackageIdentity.Id, identity.Id));
+
+                // Allowed version range for current package across all selected projects
+                var allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null);
+
+                if (allowedVersionsRange.Any())
+                {
+                    // Find a common range which satisfies all the version ranges across projects
+                    allowedVersions = VersionRange.FindCommonRange(allowedVersionsRange);
```

Comment:
yeah initially i didn't had that solution check, but now it will only have single range. will update it.